js/modules/k6/grpc/client.go

```diff
@@ -215,29 +215,21 @@ func (c *Client) Invoke(
 	ctx, cancel := context.WithTimeout(c.vu.Context(), p.Timeout)
 	defer cancel()
 
-	tags := state.Tags.GetCurrentValues()
-	tags = tags.WithTagsFromMap(p.Tags)
-
 	if state.Options.SystemTags.Has(metrics.TagURL) {
-		tags = tags.With("url", fmt.Sprintf("%s%s", c.addr, method))
+		p.Tags = p.Tags.With("url", fmt.Sprintf("%s%s", c.addr, method))
 	}
 	parts := strings.Split(method[1:], "/")
 	if state.Options.SystemTags.Has(metrics.TagService) {
-		tags = tags.With("service", parts[0])
+		p.Tags = p.Tags.With("service", parts[0])
 	}
 	if state.Options.SystemTags.Has(metrics.TagMethod) {
-		tags = tags.With("method", parts[1])
-	}
-
-	// Only set the name system tag if the user didn't explicitly set it beforehand
-	if _, ok := tags.Get("name"); !ok && state.Options.SystemTags.Has(metrics.TagName) {
-		tags = tags.With("name", method)
```

Comment:
Hmm where is this done now? he `name` tag should be set to be equal to the `url` tag, if the user didn't explicitly specify it.

Reply:
mmm...good catch, my fault. And apparently, we are not covering with tests this part.

Reply:
:disappointed: yeah...

Reply:
Adding the test I hit the issue that the URL is not set equal to the Name and I guess that is what we expect after https://github.com/grafana/k6/pull/2703 so I would prefer to add it in a dedicated bugfix PR.

Reply:
Hmm IIRC the `url` tag in gRPC has a much lower cardinality than in HTTP requests, it is just the `server/method_name`, right? So I am not sure we need to have the same "`url` tag should always be equal to the `name` tag" logic in the gRPC module that we had to add to the `k6/http` module. gRPC doesn't have any GET parameters, which is the big problem with HTTP.

Reply:
Added an issue for addressing it with a specific discussion in a issue + PR. https://github.com/grafana/k6/issues/2722